bot/events.go

```diff
@@ -3,21 +3,34 @@ package bot
 import (
 	"bytes"
 	"encoding/json"
-	"fmt"
+	"github.com/slack-go/slack/slackevents"
 	"log"
 	"net/http"
-	"strings"
-	"time"
-
-	"github.com/alecthomas/kong"
-	"github.com/bcneng/candebot/cmd"
-	"github.com/bcneng/candebot/inclusion"
-	"github.com/bcneng/candebot/slackx"
-	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
-	"github.com/slack-go/slack/slackevents"
 )
 
-func eventsAPIHandler(botCtx cmd.BotContext) http.HandlerFunc {
+// EventHandler handles a Slack event
+type EventHandler func(Context, slackevents.EventsAPIInnerEvent) error
+
+// CreateEventHandler creates a handler aware of errors (they are logged)
+func CreateEventHandler(t slackevents.EventsAPIType, f EventHandler) EventHandler {
+	return func(c Context, e slackevents.EventsAPIInnerEvent) error {
+		if slackevents.EventsAPIType(e.Type) != t {
+			log.Printf("unexpected event type. Should be %q, but was %q", t, e.Type)
```

Comment:
just wondering if we should return here

Reply:
I'm gonna remove the returned value (err) in the handler func signature since the event bus is not aware of it. Basically it does not make sense to return an error from its point of view since no one will care about.
Better to remove it and will be more clear that the error should be handled right directly in the handler.

Reply:
The event bus doesn't handle errors in the `BusSubscriber` interface: `Subscribe(topic string, fn interface{}) error`
However, I'm adding the return statement with a comment to make it consistent.